cpp/vm/evmzero/interpreter_test.cc

```diff
@@ -5006,6 +5006,36 @@ TEST(InterpreterTest, CREATE_MemoryIsGrownAlsoWithInsufficientBalance) {
   });
 }
 
+TEST(InterpreterTest, CREATE_ShanghaiLimitInitCodeSize) {
+  RunInterpreterTest({
+      .code = {op::CREATE},
+      .state_after = RunState::kErrorInitCodeSizeExceeded,
+      .gas_before = 50000,
+      .stack_before =
+          {
+              0xc001,  // < size
+              0x00,    // < offset
+              0x00,    // < value
+          },
+      .revision = EVMC_SHANGHAI,
+  });
+}
+
+TEST(InterpreterTest, CREATE_ShanghaiCodeDepositCost) {
+  RunInterpreterTest({
+      .code = {op::CREATE},
+      .state_after = RunState::kErrorGas,
+      .gas_before = 42000,  // sufficient amount of gas before Shanghai
```

Comment:
Test is called shanghai, is this before cancun? 
previous test tests shanghai as well. Is this correct? I am not familiar which exact version did introduce this change.

Reply:
This change has been introduced in Shanghai which if before Cancun.

One test is checking that the create call fails due to exceeded init code size and the other one due to insufficient gas. With this change the dynamic cost of create has been increased. The comment documents that this has been a sufficient amount of gas for this call before Shanghai. But I will rework the tests since they don't seem to be easily understandable.